### ISSUE TITLE
Add OCR text extraction job for documents, add fulltext to index, keywords in context in results

### ISSUE THREAD

**Issue by wickr** [osulp/Scholars-Archive#1408]
- [x] Add job to extract OCR text from PDFs, also Office documents (?)
  - https://github.com/samvera/hydra-derivatives/blob/master/lib/hydra/derivatives/processors/full_text.rb
  - For OD we put this in an external datastream, 'content_ocr'
- [x] Add text to search index
- [x] Add keyword in context in search results to show hits: https://oregondigital.org/sets/osu-historical-publications?utf8=%E2%9C%93&set=osu-historical-publications&q=&search_field=all_fields&set=osu-historical-publications&q=agriculture
- [ ] Allow a way to view full text on an item page? (maybe just by admins)

Some of the OD code is here:
https://github.com/OregonDigital/oregondigital/tree/master/lib/pdf_ocr
https://github.com/OregonDigital/oregondigital/blob/master/app/models/document.rb

**Comment by CGillen** [comment]
@simholt 'regarding the effects' has the color I found in the existing CSS called `highlight-yellow`, but that's only used in a dashboard badge somewhere. I think it looks good on the search and stands out, but want to get your thoughts. It passes accessibility very well.
![image](https://user-images.githubusercontent.com/11052958/109335348-c2ef2c00-7816-11eb-8b49-fade0ba7f280.png)

**Comment by wickr** [comment]
The last bullet "Allow a way to view full text on an item page? (maybe just by admins)" could likely be split out into another enhancement ticket unless Sarah really wants it. It's a 'nice to have' for me, since the only other way to see a full fulltext readout is to go to the console. This is mostly for debugging and QA type things.

**Comment by simholt** [comment]
@wickr I don't know anything about this feature -- do you mean the entire text of the item? Like a whole article would show up without having to download?

**Comment by wickr** [comment]
@simholt that's correct, all of the OCR content, probably on another page to view. At least that's what I was thinking.

**Comment by simholt** [comment]
Okay, that's fine by me for admins!

**Comment by wickr** [comment]
The latest OD2 ticket related to the showing/editing OCR part is: https://github.com/OregonDigital/OD2/issues/1513

I don't think any of this should hold up getting the OCR content in and highlighting in search results.

**Comment by CGillen** [comment]
Thanks @wickr
I agree this shouldn't hold things up. I'll split a ticket for the admin viewing part

**Comment by simholt** [comment]
I saved lorem ipsum text in PDF and Office formats, as well as two text files (not sure if these are to be included?) but the only full-text result was from a PDF deposited last year; the PDFs uploaded today did not show full-text matches, even the one that was the same file as the one that worked. ([Link to search](https://ir-staging.library.oregonstate.edu/catalog?utf8=%E2%9C%93&locale=en&search_field=all_fields&q=Vivamus))

On another set of tests, I noticed search results prioritized matches in the title, then abstract, then full-text, which is perfect.

--- 

![OCRtest](https://user-images.githubusercontent.com/33705514/109545587-e667e000-7a7d-11eb-9889-9f84fbf7e309.png)

**Comment by CGillen** [comment]
This depends on #2180 finishing. Currently that ticket is preventing the text extraction part of the ingest process from completing. Once #2180 is fixed this can be QA'd again

**Comment by simholt** [comment]
There we go -- QA pass

![Screen Shot 2021-03-04 at 1 40 09 PM](https://user-images.githubusercontent.com/33705514/110034304-24b80600-7cef-11eb-8f53-d596542392c1.png)

**Comment by CGillen** [comment]
Complete and on prod
- Allow a way to view full text on an item page? (maybe just by admins) - has been moved to a separate ticket: #2193
